chart/templates/postgres.yaml

```diff
@@ -20,11 +20,24 @@ spec:
           emptyDir:
             medium: Memory
             sizeLimit: "{{- .Values.db.shmVolume  }}"
-        - name: conf
-          configMap:
-            name: postgresql-conf
-            optional: true
       initContainers:
+        {{- if .Values.db.upgrade.perform}}
+        - image: tianon/postgres-upgrade:{{.Values.db.upgrade.old}}-to-{{.Values.db.upgrade.new}}
+          name: postgres-upgrade
+          volumeMounts:
+            - mountPath: "/current"
+              name: postgresql
+          command:
+            - sh
+            - -c
+            - "if [ ! -d '/current/postgres' ]; then exit 0; fi && rm -rf /var/lib/postgresql/{{.Values.db.upgrade.old}}/data &&
+              mv /current/postgres /var/lib/postgresql/{{.Values.db.upgrade.old}}/data && 
+              mkdir -p /var/lib/postgresql/{{.Values.db.upgrade.new}}/data &&
+              chmod -R 0750 /var/lib/postgresql/{{.Values.db.upgrade.new}}/data &&
+              chown 999:999 -R /var/lib/postgresql/{{.Values.db.upgrade.new}}/data &&
+              /usr/local/bin/docker-upgrade &&
+              mv /var/lib/postgresql/{{.Values.db.upgrade.new}}/data /current/postgres"
```

Comment:
@moshloop let me know if there's a better way to provide this script so it's more readable and I can add comments.